web/src/condense.ts

```diff
@@ -243,38 +243,23 @@ export function condense_and_collapse(elems: JQuery): void {
 
 export function initialize(): void {
     $("#message_feed_container").on("click", ".message_expander", function (this: HTMLElement, e) {
-        // Expanding a message can mean either uncollapsing or
-        // uncondensing it.
         const $row = $(this).closest(".message_row");
         const id = rows.id($row);
         assert(message_lists.current !== undefined);
         const message = message_lists.current.get(id);
         assert(message !== undefined);
-        // Focus on the expanded message.
-        message_lists.current.select_id(id);
-        const $content = $row.find(".message_content");
-        if (message.collapsed) {
-            // Uncollapse.
-            uncollapse(message);
-        } else if ($content.hasClass("condensed")) {
-            // Uncondense (show the full long message).
-            message.condensed = false;
-            uncondense_row($row);
-        }
+        toggle_collapse(message);
         e.stopPropagation();
         e.preventDefault();
     });
 
     $("#message_feed_container").on("click", ".message_condenser", function (this: HTMLElement, e) {
         const $row = $(this).closest(".message_row");
         const id = rows.id($row);
-        // Focus on the condensed message.
         assert(message_lists.current !== undefined);
-        message_lists.current.select_id(id);
         const message = message_lists.current.get(id);
         assert(message !== undefined);
-        message.condensed = true;
-        condense_row($row);
+        toggle_collapse(message);
```

Comment:
I'm having a really hard time trying to verify what impact this refactor has; here we remove selecting the message, for example; that seems like a design change, not a refactor?  And I'm not entirely sure that the state machine of condensed vs. collapsed is preserved either.

Reply:
I see the next commit adds the `select_id` back, so that part is potentially fine, though maybe would be better done as the second commit moving the `select_id`. 

But I guess a big part of what confuses me is the asymmetric model in `toggle_collapse`, where the `collapse` call is inside an `else` statement, but the `uncollapse` one is not.

Which does not seem to match the algorithm described in the comments:
``` js
    // This function implements a multi-way toggle, to try to do what                                   
    // the user wants for messages:                                                                     
    //                                                                                                  
    // * If the message is currently showing any "Show more" button, either                             
    //   because it was previously condensed or collapsed, fully display it.                            
    // * If the message is fully visible, either because it's too short to                              
    //   condense or because it's already uncondensed, collapse it                                      
                                                                                                        

```

Reply:
https://github.com/user-attachments/assets/62c2229e-4565-4512-ab7b-b75dcd21aa52

Changed the message collapse logic a bit to match what I think we wanted to implement here. Does behaviour is the screenrecording looks correct? @alya FYI.

Reply:
Note that on czo, using keyboard vs mouse results in two different behaviour when collapsing / uncollapsing messages.

Reply:
I'm pretty sure that the existing behavior was as designed; we can think about changing it, but I'd really like to avoid making UI changes like this in a PR that's trying to fix an important live bug.

Reply:
Should I just drop the last 2 commits to separate the design change here?

Reply:
Maybe; I'm not sure whether the first commit, which moves to using `toggle_collapse` for the other code path, is a behavior change... Conceptually yes, it'd be really helpful to have a pure bug fix that doesn't potentially have regressions... which might be just moving the `select_id` to the end and passing the right options?

Reply:
Okay, updated with just that.